tests/reftests/rec-pin.test

```diff
@@ -0,0 +1,994 @@
+authors: "the testing team"
+available: opam-version >= "2.1"
+bug-reports: "https://nobug"
+description: "Two words."
+dev-repo: "hg+htpps://to@lo.ck"
+homepage: "egapemoh"
+license: "ISC"
+maintainer: "maint@tain.er"
+name: "pin1sub1"
+opam-version: "2.0"
+synopsis: "A word"
+version: "2"
+x-subpath: "a"
+url {
+src: "file://${BASEDIR}/pinnes"
+}
+### opam-cat OPAM/devnull/.opam-switch/overlay/pin1sub1sub1/opam
+authors: "the testing team"
+available: opam-version >= "2.1"
+bug-reports: "https://nobug"
+description: "Two words."
+dev-repo: "hg+htpps://to@lo.ck"
+homepage: "egapemoh"
+license: "ISC"
+maintainer: "maint@tain.er"
+name: "pin1sub1sub1"
+opam-version: "2.0"
+synopsis: "A word"
+version: "2"
+x-subpath: "a/i"
+url {
+src: "git+file://${BASEDIR}/pinnes#master"
+}
+### opam unpin ./pinnes --recursive
+Ok, pin1sub1 is no longer pinned to file://${BASEDIR}/pinnes (a) (version 2)
+Ok, pin1sub1sub1 is no longer pinned to git+file://${BASEDIR}/pinnes#master (a/i) (version 2)
+Ok, pin1sub1sub1sub1 is no longer pinned to git+file://${BASEDIR}/pinnes#master (a/i/j) (version 2)
+Ok, pin1sub1sub2 is no longer pinned to file://${BASEDIR}/pinnes (a/k) (version 2)
+### opam pin scan ./pinnes | '# Name.*' -> '\c'
+pin1    2          git+file://${BASEDIR}/pinnes#master  -
+### opam pin scan ./pinnes --recursive | '# Name.*' -> '\c'
+pin1sub2          2          git+file://${BASEDIR}/pinnes#master  b
+pin1sub1sub2      2          git+file://${BASEDIR}/pinnes#master  a/k
+pin1sub1sub1sub1  2          git+file://${BASEDIR}/pinnes#master  a/i/j
+pin1sub1sub1      2          git+file://${BASEDIR}/pinnes#master  a/i
+pin1sub1          2          git+file://${BASEDIR}/pinnes#master  a
+pin1              2          git+file://${BASEDIR}/pinnes#master  -
+### opam pin scan ./pinnes --subpath a | '# Name.*' -> '\c'
+pin1sub1  2          git+file://${BASEDIR}/pinnes#master  a
+### opam pin scan ./pinnes --subpath a --recursive | '# Name.*' -> '\c'
+pin1sub1sub2      2          git+file://${BASEDIR}/pinnes#master  a/k
+pin1sub1sub1sub1  2          git+file://${BASEDIR}/pinnes#master  a/i/j
+pin1sub1sub1      2          git+file://${BASEDIR}/pinnes#master  a/i
+pin1sub1          2          git+file://${BASEDIR}/pinnes#master  a
+### opam pin scan ./pinnes --recursive --normalise
+pin1sub2.2^git+file://${BASEDIR}/pinnes#master^b
+pin1sub1sub2.2^git+file://${BASEDIR}/pinnes#master^a/k
+pin1sub1sub1sub1.2^git+file://${BASEDIR}/pinnes#master^a/i/j
+pin1sub1sub1.2^git+file://${BASEDIR}/pinnes#master^a/i
+pin1sub1.2^git+file://${BASEDIR}/pinnes#master^a
+pin1.2^git+file://${BASEDIR}/pinnes#master
+### opam pin -n --with-version 3 pin1sub2.2^git+file://${BASEDIR}/pinnes#master^b pin1sub1sub1sub1.2^git+file://${BASEDIR}/pinnes#master^a/i/j pin1sub1sub1.2^git+file://${BASEDIR}/pinnes#master^a/i pin1sub1.2^git+file://${BASEDIR}/pinnes#master^a pin1^git+file://${BASEDIR}/pinnes#master pin1sub1sub2.2^git+file://${BASEDIR}/pinnes#master^a/k
```

Comment:
I’m not sure I understand this. Is that some sort of shortcut command to subpath pin multiple things?
I’m not sure this is a desirable feature and looks more confusing than anything

Reply:
It's the `pin scan feature`, here i'm just checking that it works with subpathes